### PR TITLE
Add archer app ls command

### LINKED ISSUE #439: Would be better to have "app ls"
Currently archer does not have "app ls". As a user I have a project called "my-project". 

However, I forget what applications I have under that project and I find there's no easy way to show my applications.

### PR DESCRIPTION
**Pull request by iamhopaul123**
Fix #439. And change `archer env ls` command's prompt from "inputting project name" to "selecting projects".

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/cli/app_list.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	}
+	proj, err := opts.prompt.SelectOne(
+		applicationListProjectNamePrompt,
+		applicationListProjectNameHelper,
+		projStrs,
+	)
+	return proj, err
+}
+
+func (opts *ListAppOpts) localAppsFilter(appNames []string) {
```
Comment (efekarakus) [inline internal/pkg/cli/app_list.go]: Should we rename to `localNames` and possibly use a `map`? I got confused with all the appNames :D

```go
isLocal := make(map[string]bool)
for _, app := range opts.applications {
  if _, ok := isLocal[app.Name]; !ok {
     continue
  }
  filtered = append(filtered, app)
}
```

File: internal/pkg/cli/app_list.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+	return nil
+}
+
+func (opts *ListAppOpts) humanOutput() string {
+	b := &strings.Builder{}
+	for _, app := range opts.applications {
+		fmt.Fprintf(b, "%s: %s\n", app.Type, app.Name)
+	}
+	return b.String()
```
Comment (efekarakus) [inline internal/pkg/cli/app_list.go]: Should we remove the ":" to be consistent with the style guide (https://github.com/aws/amazon-ecs-cli-v2/blob/master/STYLE_GUIDE.md#data)

I think that's the dream look that we want to achieve with our human outputs :)

File: internal/pkg/cli/app.go [hunk 3]
```diff
@@ -25,7 +26,12 @@ An application represents an Amazon ECS service or task.`,
 		},
 	}
 
+	// The flags bound by viper are available to all sub-commands through viper.GetString({flagName})
+	cmd.PersistentFlags().StringP(projectFlag, projectFlagShort, "" /* default */, projectFlagDescription)
+	viper.BindPFlag(projectFlag, cmd.PersistentFlags().Lookup(projectFlag))
```
[the remark this answers is not in the document]
Comment (iamhopaul123) [reply internal/pkg/cli/app.go]: The --project flag will get ignored. That's a good call though! I think maybe we should not add the global flag for `app` command or we create a counterpart as a local flag instead. What do you think?

[the remark this answers is not in the document]
Comment (efekarakus) [reply internal/pkg/cli/app.go]: :+1: having --project only for `app ls` sounds good to me!

File: internal/pkg/cli/app_list.go [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+	if opts.ShouldShowLocalApps {
+		localAppManifests, err := opts.ws.Apps()
+		if err != nil {
+			return fmt.Errorf("failed to get local app manifests: %w", err)
+		}
+		var localAppNames []string
+		for _, appManifest := range localAppManifests {
+			localAppNames = append(localAppNames, appManifest.AppName())
+		}
+		opts.localAppsFilter(localAppNames)
```
Comment (sonofachamp) [inline internal/pkg/cli/app_list.go]: Workspace apps is equivalent to local apps as far as my understanding goes. What exactly are we filtering here?

Comment (sonofachamp) [reply internal/pkg/cli/app_list.go]: You can disregard, the `Apps()` function returns the manifests, not the archer Apps.

Comment (iamhopaul123) [reply internal/pkg/cli/app_list.go]: "localAppNames" provides as a filter parameter

File: internal/pkg/cli/app_list.go [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+func (opts *ListAppOpts) humanOutput() {
+	writer := tabwriter.NewWriter(opts.w, minCellWidth, tabWidth, cellPaddingWidth, paddingChar, noAdditionalFormatting)
+	fmt.Fprintf(writer, "%s\t%s\n", "Name", "Type")
+	nameLengthMax := 0
+	typeLengthMax := 0
+	for _, app := range opts.applications {
+		nameLengthMax = int(math.Max(float64(nameLengthMax), float64(len(app.Name))))
+		typeLengthMax = int(math.Max(float64(typeLengthMax), float64(len(app.Type))))
+	}
+	fmt.Fprintf(writer, "%s\t%s\n", strings.Repeat("-", nameLengthMax), strings.Repeat("-", typeLengthMax))
```
Comment (efekarakus) [inline internal/pkg/cli/app_list.go]: 🤩 omg

File: internal/pkg/cli/app_list.go [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+	}
+
+	return nil
+}
+
+func (opts *ListAppOpts) humanOutput() {
+	writer := tabwriter.NewWriter(opts.w, minCellWidth, tabWidth, cellPaddingWidth, paddingChar, noAdditionalFormatting)
+	fmt.Fprintf(writer, "%s\t%s\n", "Name", "Type")
+	nameLengthMax := 0
+	typeLengthMax := 0
```
Comment (efekarakus) [inline internal/pkg/cli/app_list.go]: maybe that should start with len("Name") and len("Type")?

Comment (iamhopaul123) [reply internal/pkg/cli/app_list.go]: :+1: you are so good!

**Pull request merged**